continuous-integration/readme.md

```diff
@@ -1,3 +1,235 @@
+
+We encourage teams to implement the CI/CD pipelines before any service code is written for customers, which usually happens in Sprint 0(N). This way, the engineering team can develop and test their work in isolation without impacting other developers and promote a consistent devops workflow throughout the engagement.
+
+These [principles](https://martinfowler.com/articles/continuousIntegration.html) map directly agile software development lifecycle [practices](https://en.wikipedia.org/wiki/Agile_software_development).
+
+<!-- markdownlint-disable MD036 -->
+**Table of contents**
+<!-- markdownlint-enable MD036 -->
+
+- [Continuous Integration](#continuous-integration)
+  - [Goals](#goals)
+  - [S/ingle Source Repository](#single-source-repository)
+  - [Build Automation](#build-automation)
+  - [Build Environment Dependencies](#build-environment-dependencies)
+  - [Infrastructure as Code](#infrastructure-as-code)
+    - [Sample DevOPS Workflow using Terraform and Cobalt](#sample-devops-workflow-using-terraform-and-cobalt)
+    - [Why](#why)
+    - [IAC DevOPS: Operations by Pull Request](#iac-devops-operations-by-pull-request)
+    - [Infrastructure Advocated Patterns](#infrastructure-advocated-patterns)
+    - [IAC Principles](#iac-principles)
+  - [Integration Validation](#integration-validation)
+  - [Git Driven Workflow](#git-driven-workflow)
+  - [Deliver Quickly and Daily](#deliver-quickly-and-daily)
+  - [Isolated Environments](#isolated-environments)
+  - [Developer Access to Latest Release Artifacts](#developer-access-to-latest-release-artifacts)
+  - [Integration Observability](#integration-observability)
+  - [Resources](#resources)
+
+## Goals
+
+Continuous integration automation is an integral part of the software development lifecycle intended to reduce build integration errors and maximize velocity across a dev crew.
+
+A robust build automation pipeline will:
+
+- Accelerate team velocity
+- Prevent integration problems
+- Avoid last minute chaos during release dates
+- Provide a quick feedback cycle for system-wide impact of local changes
+- Separate build and deployment stages
+- Measure and report metrics around build failures / success(s)
+- Increase visibility across the team enabling tighter communication
+- Reduce human errors, which is probably the most important part of automating the builds
+
+## S/ingle Source Repository
+
+- [ ] **Code / manifest artifacts required to build your project should be maintained in a single git repository.**
+
+  - CI provider-specific build [pipeline definition(s)](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops#define-pipelines-using-yaml-syntax) should reside within the mainline repository.
+
+## Build Automation
+
+An automated build should encompass the following principles:
+
+- [ ] **Build Task**
```

Comment:
If the suggestion for one repository is taken, then you may have multiple artifacts that need to be built for a given commit.